### ISSUE TITLE
Firmware upgrade supports XZ compressed image

### ISSUE THREAD

**Issue by jasaw** [ccrisan/motioneyeos#1119]
Current GZ disk images are around 70MB each for Raspberry Pis. When upgrading firmware, most of the time is spent downloading the disk image.

Using XZ compression, the same image can be compressed down to 45MB.
Example compression command:
`XZ_OPT=-9 tar cJf motioneyeos-raspberrypi.tar.xz motioneyeos-raspberrypi.img`

I don't know how much memory is required and how long it takes to extract an XZ file, but it may be a feasible option. If memory and CPU does become a concern, motionEye can always kill off motion software before starting the extract and upgrade process (not sure if it's already doing this).

Thoughts?

**Comment by ccrisan** [comment]
This is actually a very good idea. This needs to be tested and I think the firmware upgrade mechanism can be implemented to be backwards compatible.

One more little thing: the firmware upgrade mechanism, as well as many other common OS parts, have been moved to [thingOS](https://github.com/ccrisan/thingos/). This issue actually belongs to thingOS. Can we move issues across projects on github?

**Comment by jasaw** [comment]
I've created this issue as https://github.com/ccrisan/thingos/issues/4

**Comment by jasaw** [comment]
PR merged into thingos.
